scripts/exporters/gltf-exporter.js

```diff
@@ -0,0 +1,441 @@
+                        perspective: {
+                            yfov: fov * Math.PI / 180,
+                            znear: nearClip,
+                            zfar: farClip
+                        }
+                    });
+                }
+            });
+        }
+    }
+
+    writeMaterials(resources, json) {
+        if (resources.materials.length > 0) {
+            json.materials = [];
+
+            resources.materials.forEach((mat) => {
+                const name = mat.name;
+                const diffuse = mat.diffuse;
+                const emissive = mat.emissive;
+                const opacity = mat.opacity;
+                const blendType = mat.blendType;
+                const cull = mat.cull;
+
+                const material = {};
+
+                if (name && name.length > 0) {
+                    material.name = name;
+                }
+
+                if (!diffuse.equals(pc.Color.WHITE) || opacity !== 1) {
+                    material.pbrMetallicRoughness = {};
+                    material.pbrMetallicRoughness.baseColorFactor = [diffuse.r, diffuse.g, diffuse.b, opacity];
+                }
+
+                if (!emissive.equals(pc.Color.BLACK)) {
+                    material.emissiveFactor = [emissive.r, emissive.g, emissive.b];
+                }
+
+                if (blendType === pc.BLEND_NORMAL) {
+                    material.alphaMode = "BLEND";
+                }
+
+                if (cull === pc.CULLFACE_NONE) {
+                    material.doubleSided = true;
+                }
+
+                json.materials.push(material);
+            });
+        }
+    }
+
+    writeNodes(resources, json) {
+        if (resources.entities.length > 0) {
+            json.nodes = [];
+
+            resources.entities.forEach(function (entity) {
+                const name = entity.name;
+                const t = entity.getLocalPosition();
+                const r = entity.getLocalRotation();
+                const s = entity.getLocalScale();
+
+                const node = {};
+
+                if (name && name.length > 0) {
+                    node.name = name;
+                }
+
+                if (!t.equals(pc.Vec3.ZERO)) {
+                    node.translation = [t.x, t.y, t.z];
+                }
+
+                if (!r.equals(pc.Quat.IDENTITY)) {
+                    node.rotation = [r.x, r.y, r.z, r.w];
+                }
+
+                if (!s.equals(pc.Vec3.ONE)) {
+                    node.scale = [s.x, s.y, s.z];
+                }
+
+                if (entity.camera && entity.camera.enabled) {
+                    node.camera = resources.cameras.indexOf(entity.camera);
+                }
+
+                if (entity.render && entity.render.enabled) {
+                    entity.render.meshInstances.forEach((meshInstance) => {
+                        node.mesh = resources.meshInstances.indexOf(meshInstance);
+                    });
+                }
+
```

Comment:
```suggestion

                if (entity.model && entity.model.enabled) {
                    entity.model.meshInstances.forEach((meshInstance) => {
                        node.mesh = resources.meshInstances.indexOf(meshInstance);
                    });
                }
```

Write the JSON mesh id for MeshInstances inside a ModelComponent